homework-6/script-1.js

```diff
@@ -0,0 +1,83 @@
+//Написати функцію, яка приймає 4 аргументи і шукає найменше число серед них
+const getMinNumber = (num_1, num_2, num_3, num_4) => {
+    if (!isNaN(num_1) && !isNaN(num_2) && !isNaN(num_3) && !isNaN(num_4) && num_1 !== null && num_2 !== null && num_3 !== null && num_4 !== null && num_1 !== undefined && num_2 !== undefined && num_3 !== undefined && num_4 !== undefined) {
+        const result = Math.min(num_1, num_2, num_3, num_4);
+        return result;
+    } else {
+        console.log("Unfortunately, I can't figure out min number");
+    }
+}
+getMinNumber("5", 0, 6, 2);
+
+/*Написати функцію, яка приймає 2 аргументи та шукає найбільший спільний дільник для них. 
+Якщо такого числа немає, то повідомляти про відсутність*/
+const getGreatestCommonDivisor = (num_1, num_2) => {
+    if (!isNaN(num_1) && !isNaN(num_2) && num_1 !== undefined && num_2 !== undefined && num_1 !== null && num_2 !== null) {
+        if (num_2 > num_1 && num_2 > 0 && num_1 > 0) {
+            return getGreatestCommonDivisor(num_2, num_1);
+        } else if (!num_2) {
+            return num_1;
+        } else if (num_1 > num_2 && num_2 > 0 && num_1 > 0) {
+            return getGreatestCommonDivisor(num_2, num_1 % num_2);
+        } else {
+            console.log("Unfortunately,the greatest common divisor is missing");
+        }
+    } else {
+        console.log("Unfortunately,the greatest common divisor is missing");
+    }
+}
+getGreatestCommonDivisor(25, 5);
+
+//Написати функцію, яка приймає 1 аргумент та перевіряє чи дане число ідеальне
+const getPerfectNumber = (num) => {
```

Comment:
It works with strings
<img width="566" alt="image" src="https://user-images.githubusercontent.com/84010501/191563471-07922b8a-b85f-4525-b09f-641498b77a62.png">